### ISSUE TITLE
rule(MD044): Indented HTML comments don't escape the rule

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#755]
### rumdl version

0.2.40

### Installation method

mise

### Bug description

Rule MD044 (https://github.com/rvben/rumdl/blob/e06f03d515fc81054e3929b25d252020960a9c11/docs/md044.md) supports checking content inside HTML comments, but inline links and backticks can stop MD044 from flagging violations.

However, the ability to escape MD044 with inline links and backticks only works if the HTML comments start at column 1 of a line. If indented even 1 space, MD044 considers everything in the HTML comments to be prose that must adhere to MD044.

### Steps to reproduce

1. Create a file called `test.md` with the following content:
   ```markdown
   ---
   title: Level 1 heading
   ---

   <!-- For more information, see the [relevant page](test.md). -->
   <!-- For more information, see `test.md` -->
   This is a Test.

   - Level 1 item
   - Level 1 item
     <!-- For more information, see the [relevant page](test.md). -->
     <!-- For more information, see `test.md` -->
   - Level 1 item
   ```
1. Create a `.rumdl.toml` file with the following content:
   ```toml
   [global]
   enable = ["MD044"]

   [MD044]
   names = ["Test"]
   ```
1. Run `rumdl check test.md`.

### Expected behavior

Given the Markdown and configuration above, I'd expect no violations.

### Actual behavior

`rumdl` returns warnings:

```shell
test.md:11:54: [MD044] Proper name 'test' should be 'Test' [*]
test.md:12:35: [MD044] Proper name 'test' should be 'Test' [*]
```

Though identical to the earlier HTML comments, the fact they are indented makes them flagged as problems. The list around the comments doesn't matter. I've put them there to illustrate why you might want indented HTML comments.

### Configuration

```toml
[global]
enable = ["MD044"]

[MD044]
names = ["Test"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.2.41.

MD044 was computing its "inside an HTML comment" check from byte offsets anchored at column 0, so a comment indented even one space fell outside those bounds and its contents were treated as ordinary prose. The check now uses the trimmed content bounds, so indented comments are recognized the same as column-0 ones, and inline links and backticks inside them escape the rule regardless of indentation.

Your exact reproduction now reports no violations on v0.2.41:

```
$ rumdl check test.md
Success: No issues found in 1 file
```

Thanks for the clear repro. Please reopen if you still run into this.
